portal/server/routes/middleware/virusScanUpload/index.js

```diff
@@ -0,0 +1,78 @@
+const NodeClam = require('clamscan');
+const { Readable } = require('stream');
+
+const { CLAMAV_HOST, CLAMAV_PORT, CLAMAV_DEBUG_MODE_ENABLED, CLAMAV_SCANNING_ENABLED } = process.env;
+
+const virusScanUpload = async (req, res, next) => {
+  if (CLAMAV_SCANNING_ENABLED && req.file?.buffer) {
+    try {
+      const ClamScan = await new NodeClam()
+        .init({
+          removeInfected: false, // If true, removes infected files
+          quarantineInfected: false, // False: Don't quarantine, Path: Moves files to this place.
+          debugMode: CLAMAV_DEBUG_MODE_ENABLED, // Whether or not to log info/debug/error msgs to the console
+          scanRecursively: true, // If true, deep scan folders recursively
+          clamdscan: {
+            host: CLAMAV_HOST, // IP of host to connect to TCP interface
+            port: CLAMAV_PORT, // Port of host to use when connecting via TCP interface
+            timeout: 15000, // Timeout for scanning files
+            multiscan: true,
+          },
+          preference: 'clamdscan', // If clamdscan is found and active, it will be used by default
+        })
+        .catch((err) => {
+          res.locals.virusScanFailed = true;
+          console.error('Could not connect to clamav server');
+          console.error(err);
+          return next();
+        });
+      // can't do clamscan.scanstream when hitting remote server due to https://github.com/kylefarris/clamscan/issues/101
+      // so we use passthrough instead
+      const scanResult = await new Promise((resolve, reject) => {
+        const inputStream = Readable.from(req.file.buffer);
+        const clamAVStream = ClamScan.passthrough();
+        inputStream.pipe(clamAVStream);
+        clamAVStream
+          .on('scan-complete', (result) => {
+            const infected = result.isInfected;
+            const { viruses } = result;
+            if (infected !== null) {
+              console.error(`Scan complete; contents infected: ${infected} - ${viruses}`);
+              resolve({ infected, viruses });
+            }
+          })
+          .on('error', (error) => {
+            console.error('Clamav virus scan errored with: %O', error);
+            reject(error);
+          })
+          .on('timeout', (error) => {
+            const timeoutError = error || new Error('Scan timed out');
+            console.error('Clamav virus scan timed out');
+            reject(timeoutError);
+          });
+      });
+
+      if (scanResult?.isInfected === true) {
```

Comment:
Can this just be 
```suggestion
      if (scanResult?.isInfected) {
```

Reply:
Thought about this - I don't really know what the package returns here. It's supposed to be a boolean but sometimes its null/undefined when it errors so I'm worried about it returning something dumb like a string and me treating it as true when I shouldn't. Pure paranoia but wanted to be safe as possible

Reply:
Good shout, I'll do it in my other PR so I can merge this, busy afternoon :D